### DISCUSSION TITLE
What is the best way to hash a shaped array?

### DISCUSSION THREAD

**Discussion by DechinPhy** [jax-ml/jax#8352] [Q&A]
For example we can use `hash(ndarray.tostring())` to get the hash value of a numpy array. So what could be the best way to hash a shaped array in jax?

**Comment by jakevdp** [comment]
It depends on the context, but something like `hash(np.asarray(x).tostring())` should work.

**Reply by PhilipVinc** [reply]
better to use `.tobytes()`  , as `tostring()` is deprecated..
